src/Vector3.h

```diff
@@ -393,11 +393,9 @@ class VectorHasher
 	{
 		// Guaranteed to have no hash collisions for any 128x128x128 area
 		size_t Hash = 0;
-		Hash ^= static_cast<size_t>(a_Vector.x);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.y);
-		Hash <<= 8;
-		Hash ^= static_cast<size_t>(a_Vector.z);
+		Hash ^= a_Vector.x + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.y + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
+		Hash ^= a_Vector.z + 0x9e3779b9 + (Hash << 6) + (Hash >> 2);
```

Comment:
- The old hash was good for everything. And it's only used in the pathfinder.
 - I thought this was about chunk hashing and that confused me, since chunks have no Y.
 - Right now, I have no clue why this change was made, since vectors aren't used in the PR.